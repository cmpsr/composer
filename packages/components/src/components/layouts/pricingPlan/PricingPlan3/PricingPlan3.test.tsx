import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { PricingPlan3, PRICING_PLAN_DEFAULT_TEST_ID } from '.';

describe('PricingPlan3', () => {
  const title = 'test';
  const price = '10';
  const items = ['test 1', 'test 2', 'test 3', 'test 4'];
  const button = <a>test</a>;

  it('should match snapshot', () => {
    const pricingPlan = renderer.create(
      <PricingPlan3 title={title} price={price}>
        {{ listItems: items, button: button }}
      </PricingPlan3>,
    );
    const tree = pricingPlan.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(
      <PricingPlan3 title={title} price={price} />,
    );
    const pricingPlan = getByTestId(PRICING_PLAN_DEFAULT_TEST_ID);
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render className', () => {
    const { getByTestId } = render(
      <PricingPlan3 title={title} price={price} className="foo" />,
    );
    const pricingPlan = getByTestId(PRICING_PLAN_DEFAULT_TEST_ID);
    expect(pricingPlan.classList).toContain('foo');
  });
  it('should render title', () => {
    const { getByText } = render(<PricingPlan3 title={title} price={price} />);
    const pricingPlan = getByText('test');
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render price', () => {
    const { getByText } = render(<PricingPlan3 title={title} price={price} />);
    const pricingPlan = getByText('$10');
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render children list in two columns', () => {
    const { getAllByText } = render(
      <PricingPlan3 title={title} price={price}>
        {{ listItems: items }}
      </PricingPlan3>,
    );

    const pricingPlan = getAllByText('test 1');
    expect(pricingPlan.length).toBe(2);
  });
  it('should render children button', () => {
    const childrenTestId = 'Children-TestId';
    const link = <a data-testid={childrenTestId}>test</a>;
    const { getByTestId } = render(
      <PricingPlan3 title={title} price={price}>
        {{ listItems: items, button: link }}
      </PricingPlan3>,
    );
    const pricingPlan = getByTestId(childrenTestId);
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should call clickedMore features on click', () => {
    const { getByTestId } = render(
      <PricingPlan3 title={title} price={price}>
        {{ listItems: items }}
      </PricingPlan3>,
    );
    const button = getByTestId('button');
    fireEvent.click(button);
    expect(button.classList).toContain('hidden');
  });
  it('should render component when items is empty', () => {
    const { getByTestId } = render(
      <PricingPlan3 title={title} price={price}>
        {{ listItems: null }}
      </PricingPlan3>,
    );

    const pricingPlan = getByTestId(PRICING_PLAN_DEFAULT_TEST_ID);
    expect(pricingPlan).toBeInTheDocument();
  });
});
