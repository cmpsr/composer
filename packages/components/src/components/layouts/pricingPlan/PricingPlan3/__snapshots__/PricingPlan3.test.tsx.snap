// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingPlan3 should match snapshot 1`] = `
<div
  className="flex flex-col md:flex-row md:justify-between items-center md:px-6 py-8"
  data-testid="pricingPlan3"
>
  <div
    className="flex min-w-full justify-between items-center py-8 border-t border-b border-gray-700 md:flex-col md:min-w-0 md:items-start md:py-0 md:border-0"
  >
    <h5
      className="typo-headline-6"
      data-testid="typography"
    >
      test
    </h5>
    <div
      className="flex flex-row items-baseline"
    >
      <span
        className="typo-headline-1 font-bold"
        data-testid="typography"
      >
        $
        10
      </span>
      <span
        className="typo-body ml-2"
        data-testid="typography"
      >
        / Mo
      </span>
    </div>
  </div>
  <hr
    className="md:border md:h-20"
  />
  <div
    className="hidden md:flex justify-start min-w-full md:px-4 md:py-8  md:min-w-0 lg:p-8"
  >
    <ul>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 1
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 2
        </span>
      </li>
    </ul>
    <ul>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 3
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 4
        </span>
      </li>
    </ul>
  </div>
  <div
    className="md:hidden justify-start min-w-full p-8 md:min-w-0 "
  >
    <ul>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 1
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 2
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 3
        </span>
      </li>
      <li
        className="mb-3 hidden"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 4
        </span>
      </li>
    </ul>
    <button
      className="flex focus:outline-none"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="mr-5"
        data-testid="icon"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
      >
        <path
          d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
      View all features
    </button>
  </div>
  <div
    className="min-w-full md:min-w-0"
  >
    <a>
      test
    </a>
  </div>
</div>
`;
